javacord-api/src/main/java/org/javacord/api/entity/Nameable.java

```diff
@@ -1,14 +1,38 @@
 package org.javacord.api.entity;
 
+import org.javacord.api.util.FormatUtil;
+import org.javacord.api.util.logging.ExceptionLogger;
+
+import java.io.IOException;
+import java.util.Formattable;
+import java.util.FormattableFlags;
+import java.util.Formatter;
+
 /**
  * This class represents an entity which has a name.
+ *
+ * <p>If used in a format string, a {@code Nameable} will be rendered as
+ * the entity's name unless the alternate flag is given.
+ *
+ * @see String#format(String, Object...)
  */
-public interface Nameable {
+public interface Nameable extends Formattable {
 
     /**
      * Gets the name of the entity.
      *
      * @return The name of the entity.
      */
     String getName();
+
+    @Override
+    default void formatTo(Formatter formatter, int flags, int width, int precision) {
+        boolean alternate = (flags & FormattableFlags.ALTERNATE) != 0;
+        String s = alternate ? this.toString() : this.getName();
+        try {
+            FormatUtil.format(s, formatter, flags, width);
+        } catch (IOException e) {
+            ExceptionLogger.get().apply(e);
```

Comment:
That's not how the ExceptionLogger is meant to be used.
I can see why it's tempting to just create a default method in the api module, but to access the proper log4j2 logger, the implementation must be moved into the core module.

Reply:
As there are `Nameable` enums in the API part, it is hard to move the implementation into core as mentioned in the comment thread below.
So it might actually be ok to use the exception logger here.
I'd just use instead of `ExceptionLogger.get().apply(e);`
```suggestion
            ExceptionLogger.getConsumer().accept(e);
```
as it has the better semantic here to use the `Consumer` variant, not the `Function` variant as you don't use the return value.

Actually I also used this construct in `org.javacord.api.entity.channel.TextChannel#typeContinuouslyAfter(long, java.util.concurrent.TimeUnit, java.util.function.Consumer<java.lang.Throwable>)` and there you accepted it. I don't remember though why I didn't move the implementation to the core package. Maybe there was no `InternalTextChannel` interface yet and I didn't want to duplicate the code. Or I wanted the unwrap and enrich logic we provide in the exception logger.

Anyway, if we keep the code in API because of the two enums, I'd say it is ok to here use the exception logger, especially as it should most probably not be triggered anywy, but in the variant I mentioned above.